src/main/java/spiffe/api/svid/WorkloadAPIClient.java

```diff
@@ -0,0 +1,69 @@
+package spiffe.api.svid;
+
+import io.grpc.stub.StreamObserver;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.List;
+import java.util.function.Consumer;
+
+import static org.apache.commons.lang3.StringUtils.EMPTY;
+import static spiffe.api.svid.Workload.*;
+
+/**
+ * Provides functionality to interact with a Workload API
+ *
+ */
+public final class WorkloadAPIClient {
+
+    private static Logger LOGGER = LoggerFactory.getLogger(WorkloadAPIClient.class);
+
+    private SpiffeWorkloadStub spiffeWorkloadStub;
+
+    /**
+     * Constructor
+     * @param spiffeEndpointAddress
+     */
+    public WorkloadAPIClient(String spiffeEndpointAddress) {
+        spiffeWorkloadStub = new SpiffeWorkloadStub(spiffeEndpointAddress);
+    }
+
+    /**
+     * Default constructor
+     * The WorkloadAPI Address will be resolved by an Environment Variable
+     *
+     */
+    public WorkloadAPIClient() {
+        spiffeWorkloadStub = new SpiffeWorkloadStub(EMPTY);
```

Comment:
IMO Determining the configuration for the connection should be done on the outside of connection handler itself to separate any configuration management code from the business logic.